tests/utils/test_data_utils.py

```diff
@@ -0,0 +1,66 @@
+# Copyright The FMS HF Tuning Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# SPDX-License-Identifier: Apache-2.0
+# https://spdx.dev/learn/handling-license-info/
+
+# Third Party
+import datasets
+import pytest
+
+# First Party
+from tests.data import TWITTER_COMPLAINTS_DATA
+
+# Local
+from tuning.utils import data_utils
+
+
+def test_apply_custom_formatting_template():
+    json_dataset = datasets.load_dataset("json", data_files=TWITTER_COMPLAINTS_DATA)
+    template = "### Input: {{Tweet text}} \n\n ### Response: {{text_label}}"
```

Comment:
Some overlap with previous comment on whether or not we want to add more restrictions or at least recommended conventions for templates here, since these are a bit different 😄